tests/Unit/Parallel/Test_PupStlCpp17.cpp

```diff
@@ -10,6 +10,24 @@
 #include "Framework/TestHelpers.hpp"
 #include "Parallel/PupStlCpp17.hpp"
 
+namespace {
+struct LocalNonCopyable {
+  LocalNonCopyable() = default;
+  explicit LocalNonCopyable(const size_t in_t) : t(in_t) {}
+  LocalNonCopyable(const LocalNonCopyable&) = delete;
+  LocalNonCopyable& operator=(const LocalNonCopyable&) = delete;
+  LocalNonCopyable(LocalNonCopyable&&) = default;
```

Comment:
If the problem only occurs when the class is neither copyable nor movable, does this test actually fail with the previous implementation? If not, I think it should be made non-movable also, to truly test the functionality you've implemented.

Reply:
I agree, but I can't test that without an in-place deserialize like you are adding in the global cache serialization PR. A few options:
1. I grab that commit from your PR and add it here
2. You factor out that commit (I'm not yet convinced of the rest of the changes in that PR, need to think about it more and probably would be good to have a discussion)
3. This blocks until the global cache PR is merged.

I would generally vote 1. or 2. (indifferent on which one). The reason I split out the several small PRs is that others might need them and they are easier to review and merge than all the other commits I have on the branch (i.e. avoid blocking). Let me know what you would prefer

Reply:
Okay, I split out that simple commit to #2623

Reply:
I'll rebase and ping you once that's merged :)